demo/apple/QuranSyncDemo/QuranSyncDemo/BookmarkViewModel.swift

```diff
@@ -5,39 +5,72 @@
 //  Created by Claude AI on 5/18/25.
 //
 
+import Combine
 import Foundation
 import Shared
-import Combine
 
+@MainActor
 class BookmarkViewModel: ObservableObject {
-  @Published var bookmarks: [Page_bookmark] = []
+  struct BookmarkItem: Identifiable, Equatable {
+    let id: String
+    let page: Int
+    let lastUpdated: Date
+  }
+  
+  @Published var bookmarks: [BookmarkItem] = []
   
   private let databaseManager = DatabaseManager.shared
+  private var bookmarksTask: Task<Void, Never>?
   
   init() {
-    loadBookmarks()
+    observeBookmarks()
   }
   
-  func loadBookmarks() {
-    bookmarks = databaseManager.getPageBookmarks()
+  deinit {
+    bookmarksTask?.cancel()
+  }
+  
+  private func observeBookmarks() {
+    bookmarksTask?.cancel()
+    bookmarksTask = Task {
+      do {
+        let sequence = databaseManager.bookmarksSequence()
+        for try await sharedBookmarks in sequence {
+          bookmarks = sharedBookmarks.map(Self.mapToItem)
+        }
+      } catch {
+        print("Failed to observe bookmarks: \(error)")
+      }
+    }
   }
   
   func addRandomBookmark() {
-    databaseManager.addRandomBookmark()
-    loadBookmarks() // Refresh the list after adding
```

Comment:
unlike before, we no longer have to refresh the list after adding because it's automagically updated due to listening to the flow